alf/environments/suite_bsuite.py

```diff
@@ -0,0 +1,106 @@
+        env_id (int): (optional) ID of the environment.
+        discount (float): Discount to use for the environment.
+        max_episode_steps (int): If None the max_episode_steps will be set to zero as not
+        all bsuite environments specify max episode lengths.
+            if set to 0 or if there is no max_episode_steps set in the environment's
+            spec.
+        gym_env_wrappers (Iterable): Iterable with references to gym_wrappers
+            classes to use directly on the gym environment.
+        alf_env_wrappers (Iterable): Iterable with references to alf_wrappers
+            classes to use on the ALF environment.
+
+    Returns:
+        An AlfEnvironment instance.
+    """
+
+    env = bsuite.load_from_id(environment)
+    gym_env = BSuiteWrapper(env)
+
+    if max_episode_steps is None:
+        max_episode_steps = 0
+
+    return wrap_env(
+        gym_env,
+        env_id=env_id,
+        discount=discount,
+        max_episode_steps=max_episode_steps,
+        gym_env_wrappers=gym_env_wrappers,
+        alf_env_wrappers=alf_env_wrappers,
+        image_channel_first=False)
+
+
+class BSuiteWrapper(gym_wrapper.GymFromDMEnv):
+    """A wrapper for Bsuite environment.
+
+    The BSuite environment is introduced in
+    `Osband et al. Behaviour Suite for Reinforcement Learning  <https://openreview.net/forum?id=rygf-kSYwH>`_.
+
+    It can be accessed on https://github.com/deepmind/bsuite
+    """
+
+    _GymTimestep = Tuple[np.ndarray, float, bool, Dict[str, Any]]
+
+    def __init__(self, env):
+        """
+        Args:
+            gym_env (gym.Env): An instance of OpenAI gym environment.
+        """
+        super(BSuiteWrapper, self).__init__(env)
+
+    @property
+    def observation_space(self) -> spaces.Box:
+        obs_spec = self._env.observation_spec()  # type: specs.Array
+        obs_spec = specs.Array(
+            shape=(obs_spec.shape[1],), dtype=np.float32, name='state')
+        if isinstance(obs_spec, specs.BoundedArray):
+            return spaces.Box(
+                low=float(obs_spec.minimum),
+                high=float(obs_spec.maximum),
+                shape=obs_spec.shape,
+                dtype=obs_spec.dtype)
+        return spaces.Box(
+            low=-float('inf'),
+            high=float('inf'),
+            shape=obs_spec.shape,
+            dtype=obs_spec.dtype)
+
+    def step(self, action: int) -> _GymTimestep:
+        timestep = self._env.step(action)
+        self._last_observation = timestep.observation
+        reward = timestep.reward or 0.
+        if timestep.last():
+            self.game_over = True
+        return np.reshape(timestep.observation, (timestep.observation.shape[1], )), reward, timestep.last(), {}
+
+    def reset(self) -> np.ndarray:
+        self.game_over = False
+        timestep = self._env.reset()
+        self._last_observation = timestep.observation
+        return np.reshape(timestep.observation, (timestep.observation.shape[1], ))
```

Comment:
A simple test file would be needed. See ``suite_safety_gym_test.py`` or ``suite_babyai_test.py`` for examples.

Reply:
Added suite_bsuite_test.py for some simple tests.